projects/plugins/jetpack/extensions/blocks/story/story.php

```diff
@@ -248,38 +268,49 @@ class="wp-story-video intrinsic-ignore wp-video-%3$s"
 }
 
 /**
- * Render a slide
+ * Pick a thumbnail to render a static/embedded story
  *
- * @param array $media  - Media information.
- * @param int   $index  - Index of the slide, first slide will be displayed by default, others hidden.
+ * @param array $media_files  - list of Media files.
  *
  * @returns string
  */
-function render_slide( $media, $index = 0 ) {
+function render_static_slide( $media_files ) {
 	$media_template = '';
-	$media_type     = ! empty( $media['type'] ) ? $media['type'] : null;
-	if ( ! $media_type ) {
+	if ( empty( $media_files ) ) {
 		return '';
 	}
-	switch ( $media_type ) {
-		case 'image':
-			$media_template = render_image( $media, $index );
-			break;
-		case 'video':
-			$media_template = render_video( $media, $index );
-			break;
-		case 'file':
-			// VideoPress videos can sometimes have type 'file', and mime 'video/videopress' or 'video/mp4'.
-			if ( 'video' === substr( $media['mime'], 0, 5 ) ) {
-				$media_template = render_video( $media, $index );
+
+	// find an image to showcase.
+	foreach ( $media_files as $media ) {
+		switch ( $media['type'] ) {
+			case 'image':
+				$media_template = render_image( $media );
+				break 2;
+			case 'video':
+				// ignore videos without a poster image.
+				if ( empty( $media['poster'] ) ) {
+					continue 2;
+				}
+				$media_template = render_video( $media );
+				break 2;
+		}
+	}
+
+	// if no media was found try to render a video tag without poster.
+	if ( empty( $media_template ) ) {
+		foreach ( $media_files as $media ) {
+			switch ( $media['type'] ) {
+				case 'video':
+					$media_template = render_video( $media );
```

Comment:
I'm not sure I understand why we need to loop again here instead of adding this above? I must be missing something obvious, but maybe that's worth explaining in a comment (or just here to me :) )

Reply:
Sure, here's the deal:

We're trying to find a thumbnail image for the story, so we can render it statically for email notifications or RSS readers for instance.

The logic is simple:
- First loop: We go over each slide, if there's an image or a video with a poster we use that slide for the thumbnail (in the case of a video, its poster is used) and we break out of the loop (`break 2;`).
- Second loop: it only runs if we haven't found a thumbnail for the story yet (`empty( $media_template )`). Which means this story probably only has videos without a poster. So we take the first video slide we encounter (most likely the first one) and use it as the thumbnail. Some environment will be able to render the video and load it even without JS (for instance a browser with JS disabled such as the WordPress.com reader) but others won't (for instance an email client).

I think we could get rid of the second loop here and just pick the first slide. I find it a bit more logical to use loops as it handles a no slide story as well and maps easily to functional code in my head. iirc I tried using `array_filter` at first but PHP made it clearer to just use loops.

The comment ontop of this second loop should be explaining this `// if no media was found try to render a video tag without poster.`, I'm not sure how to make it clearer, if you have some thoughts?

Reply:
Done in https://github.com/Automattic/jetpack/pull/19876/commits/6a3326f4e0190ad4a01fcf09983318e120fe9db9